backend/import_status/get.py

```diff
@@ -57,13 +58,40 @@ def get_import_status(event: JsonObject) -> JsonObject:
         )
 
     response_body = {
-        "validation": {"status": step_function_resp["status"]},
+        "validation": {
+            "status": step_function_resp["status"],
+            "errors": get_step_function_validation_results(step_function_resp["input"]),
+        },
         "upload": upload_response,
     }
 
     return success_response(200, response_body)
 
 
+def get_step_function_validation_results(step_function_input: str) -> JsonList:
+    input_json = json.loads(step_function_input)
+    hash_key = f"DATASET#{input_json['dataset_id']}#VERSION#{input_json['version_id']}"
```

Comment:
Generally it's considered bad practice to pass an unparsed object to a function and letting it parse it, because it makes the implementation of the inner function rely on the structure of an object retrieved by the outer function. I'd rather pass the dataset and version IDs to this function.